src/image-copy/azext_imagecopy/create_target.py

```diff
@@ -113,11 +116,21 @@ def create_target_image(location, transient_resource_group_name, source_type, so
     else:
         snapshot_resource_group_name = transient_resource_group_name
 
+    storage_account_name = target_blob_path.split('.')[0].split('/')[-1]
+    if target_subscription:
+        subscription_id = target_subscription
+    else:
+        subscription_id = get_subscription_id(cmd.cli_ctx)

```

Comment:
is this original service behavior? otherwise, pls help to document the behavior.

Reply:
The goal of this PR is to infer storage account ID. If target_subscription is None, use the default subscription.
Code that creates storage account.
```
cli_cmd = prepare_cli_command(['storage', 'account', 'create',
                                   '--name', target_storage_account_name,
                                   '--resource-group', transient_resource_group_name,
                                   '--location', location,
                                   '--sku', 'Standard_LRS'],
                                  subscription=target_subscription)
```